### PR TITLE
Implement ON CLUSTER syntax when we want to reload a dictionary

### PR DESCRIPTION
**Pull request by YiuRULE**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- New Feature

Changelog entry (up to few sentences, required except for Non-significant/Documentation categories):

Implement ON CLUSTER feature for SYSTEM RELOAD DICTIONARY

Detailed description (optional):

Right now, we cannot reload a dictionary on a distributed environment, so when we have an update from the dictionary source and we try to reload the dictionary, we need to do it instance by instance, who is not really convenient and can put some problems between replicas where the returned rows could be different depending if the dictionary has been reloaded or not.

This PR has a goal to implement the `ON CLUSTER` syntax for the command `SYSTEM RELOAD DICTIONARY`, in a similar fashion at `SYSTEM RELOAD DICTIONARY ON CLUSTER 'cluster' dict_name`, where this command will run this command to every shards and replicas of the cluster.

### CODE REVIEW

**Comment by YiuRULE** [comment]
Test concerning the `SYSTEM RELOAD DICTIONARY ON CLUSTER` has been fixed and resolve the merge conflict.

The tests who currently failed are not related to the new feature

**Review by alexey-milovidov — APPROVED** [review]

File: dbms/tests/integration/test_dictionary_ddl_on_cluster/test.py [hunk 1]
```diff
@@ -55,6 +56,16 @@ def test_dictionary_ddl_on_cluster(started_cluster):
         assert node.query("SELECT count() from sometbl") == "1\n"
         assert node.query("SELECT dictGetString('default.somedict', 'value', toUInt64({}))".format(num)) == node.name + '\n'
 
+
+    for num, node in enumerate([ch1, ch2, ch3, ch4]):
+        node.query("ALTER TABLE sometbl UPDATE value = 'new_key' WHERE 1")
+
+    ch1.query("SYSTEM RELOAD DICTIONARY ON CLUSTER 'cluster' `default.somedict`")
+    time.sleep(2)  # SYSTEM RELOAD DICTIONARY is an asynchronous query
```
Comment (azat) [inline dbms/tests/integration/test_dictionary_ddl_on_cluster/test.py]: it's not already, since #8240, so `sleep` can be dropped

Comment (alexey-milovidov) [reply dbms/tests/integration/test_dictionary_ddl_on_cluster/test.py]: b17048f07a

**Comment by filimonov** [comment]
https://github.com/ClickHouse/ClickHouse/issues/6418

**Pull request merged**